src/client/components/pages/Courses/MeetingTimesList.tsx

```diff
@@ -0,0 +1,457 @@
+    justify-self: end;
+    > * {
+      margin-left: ${fromTheme('ws', 'xsmall')};
+    }
+`;
+
+/**
+ * The meeting times section of the modal that allows users to edit meetings
+ * for a particular course instance.
+ */
+export const MeetingTimesList
+: FunctionComponent<MeetingTimesListProps> = function ({
+  meetings,
+  saving,
+  onChange,
+}): ReactElement {
+  /**
+   * The meeting within the list of meetings that is currently being edited
+   */
+  const [
+    currentMeetingId,
+    setCurrentMeetingId,
+  ] = useState(null as string);
+
+  /**
+   * The selected day in the dropdown for the meeting currently being edited
+   */
+  const [
+    currentDay,
+    setCurrentDay,
+  ] = useState(null as DAY);
+
+  /**
+   * The start time value for the meeting currently being edited
+   */
+  const [
+    currentStartTime,
+    setCurrentStartTime,
+  ] = useState(null as string);
+
+  /**
+   * The end time value for the meeting currently being edited
+   */
+  const [
+    currentEndTime,
+    setCurrentEndTime,
+  ] = useState(null as string);
+
+  /**
+   * The room assigned to the meeting currently being edited
+   */
+  const [
+    currentRoom,
+    setCurrentRoom,
+  ] = useState(null as MeetingRoomResponse);
+
+  /**
+   * The current value of the error message when creating or editing a meeting time
+   */
+  const [
+    meetingTimeError,
+    setMeetingTimeError,
+  ] = useState('');
+
+  /**
+   * Used to create a temporary unique ID for new meetings on the client. A permanent UUID will be assigned as a result of the server request
+   */
+  const [
+    newMeetingIndex,
+    setNewMeetingIndex,
+  ] = useState(0);
+
+  /**
+   * The fields of the existing meeting are checked to make sure they are non-empty.
+   * The entered start and end times are compared to make sure the start time is
+   * not later than the end time. In either case, an error is set.
+   */
+  const validateTimes = (): boolean => {
+    if (currentMeetingId) {
+      if (currentDay === '' as DAY || currentStartTime === '' || currentEndTime === '') {
+        setMeetingTimeError('Please provide a day and start/end times before proceeding.');
+        return false;
+      }
+      if (currentStartTime >= currentEndTime) {
+        setMeetingTimeError('End time must be later than start time.');
+        return false;
+      }
+    }
+    setMeetingTimeError('');
+    return true;
+  };
+
+  useEffect(() => {
+    if (saving) {
+      validateTimes();
+    }
+  });
+
+  /**
+   * Retrieves and is used to update the current list of meetings, which lives
+   * in the state of the parent component (MeetingModal)
+   */
+  const getCurrentMeetings = () => meetings
+    .map((meeting: CourseInstanceResponseMeeting) => (
+      meeting.id === currentMeetingId
+        ? {
+          ...meeting,
+          day: currentDay,
+          startTime: currentStartTime,
+          endTime: currentEndTime,
+          room: currentRoom,
+        }
+        : meeting));
```

Comment:
I'm still a little concerned with this approach. It feels like there's a lot of state that still lives in the `MeetingTimesList`, and that makes me a little concerned about updates either being overwritten or not percolating down when changes are made.

That said, I think this is OK in the context of this pull request, but I'd like to revisit this before we implement more features on top of it.